HACKING.md

```diff
@@ -1,114 +1,233 @@
-to 2.52, snapd requires Go 1.9+. Versions before 2.38 support Go 1.6+.
+Go 1.13 or later is required to build `snapd`.
 
-### Setting up a GOPATH
+If you need to build older versions of snapd, please have a look at the file
+`debian/control` to find out what dependencies were needed at the time
+(including which version of the Go compiler).
+
+### Setting up your build environment
+
+If your Go environment (e.g. `GOPATH`) is already configured, you should skip
+this step. The Go environment setup can be added to your shell login script
+(e.g. ~/.bashrc for bash) for automatic setup (after a terminal restart).
+
+```
+export GOPATH=${HOME}/work
+mkdir -p $GOPATH
+export PATH="$PATH:$GOPATH/bin"
+```
+
+### Further environment variable details
 
 When working with the source of Go programs, you should define a path within
 your home directory (or other workspace) which will be your `GOPATH`. `GOPATH`
-is similar to Java's `CLASSPATH` or Python's `~/.local`. `GOPATH` is documented
-[online](http://golang.org/pkg/go/build/) and inside the go tool itself
+is similar to Java's `CLASSPATH` or Python's `~/.local`. `GOPATH` is
+documented [online](http://golang.org/pkg/go/build/) and inside the `go` tool
+itself.
 
     go help gopath
 
 Various conventions exist for naming the location of your `GOPATH`, but it
-should exist, and be writable by you. For example
+should exist, and be writable by you. For example:
 
     export GOPATH=${HOME}/work
     mkdir $GOPATH
 
 will define and create `$HOME/work` as your local `GOPATH`. The `go` tool
 itself will create three subdirectories inside your `GOPATH` when required;
-`src`, `pkg` and `bin`, which hold the source of Go programs, compiled packages
-and compiled binaries, respectively.
+`src`, `pkg` and `bin`, which hold the source of Go programs, compiled
+packages and compiled binaries, respectively.
 
 Setting `GOPATH` correctly is critical when developing Go programs. Set and
 export it as part of your login script.
 
-Add `$GOPATH/bin` to your `PATH`, so you can run the go programs you install:
+Add `$GOPATH/bin` to your `PATH`, so you can run the Go programs you install:
 
     PATH="$PATH:$GOPATH/bin"
 
 (note `$GOPATH` can actually point to multiple locations, like `$PATH`, so if
 your `$GOPATH` is more complex than a single entry you'll need to adjust the
 above).
 
-Note that if you are using go 1.16 or newer you need to disable the
-go modules feature. Use:
 
-    export GO111MODULE=off
+### Getting the snapd sources
 
-for this.
+The easiest way to get the source for `snapd` is to clone the GitHub repository
+in a directory where you have read-write permissions, such as your home
+directory.
 
+    cd ~/
+    git clone https://github.com/snapcore/snapd.git
+    cd snapd
 
-### Getting the snapd sources
+This will allow you to build and test `snapd`. If you wish to contribute to
+the `snapd` project, please see the Contributing section.
+
+### Contributing
```

Comment:
I would keep this section after the "Building" one, since one typically won't start contributing before building (and most people will only be interested in building, not contributing).